src/main/functions/DidJWT.ts

```diff
@@ -191,7 +192,7 @@ export function getAudience(jwt: string) {
 }
 
 function assertIssSelfIssuedOrDid(payload: JWTPayload) {
-  if (!payload.iss || !(payload.iss.startsWith('did:') || isIssSelfIssued(payload))) {
+  if (!payload.iss || !(payload.sub.startsWith('did:') || isIssSelfIssued(payload))) {
```

Comment:
According to [this](https://openid.net/specs/openid-connect-self-issued-v2-1_0-ID1.html#name-self-issued-id-token) it can be a did or jwk, considering that we are talking about the id token

Reply:
Yeah but the name of the function references ISS (so issuer). So it should be about sub altogether with a name change of the method, or it should be about iss. I am doing the review from Github, so don't have the full context where this is being called from. Probably from creating the AuthResponse IDToken. In which case it indeed should become sub, but then also the first part should be changed to sub

Reply:
In the JWT request it is [iss](https://openid.net/specs/openid-connect-federation-1_0.html#name-authentication-request) and in the id token JWT in the response it is [sub](https://openid.net/specs/openid-connect-self-issued-v2-1_0-ID1.html#name-self-issued-id-token). Since it's used only in the response, I'm assuming it needs to be sub. Put a TODO on it because I think it needs to be revisited to support automatic registration.